core/src/main/java/de/danoeh/antennapod/core/storage/PodDBAdapter.java

```diff
@@ -1233,14 +1237,27 @@ public Cursor searchItems(long feedID, String searchQuery) {
             queryFeedId = "1 = 1";
         }
 
-        String query = SELECT_FEED_ITEMS_AND_MEDIA_WITH_DESCRIPTION
-                + " WHERE " + queryFeedId + " AND ("
-                + KEY_DESCRIPTION + " LIKE '%" + preparedQuery + "%' OR "
-                + KEY_CONTENT_ENCODED + " LIKE '%" + preparedQuery + "%' OR "
-                + KEY_TITLE + " LIKE '%" + preparedQuery + "%'"
-                + ") ORDER BY " + KEY_PUBDATE + " DESC "
-                + "LIMIT 300";
-        return db.rawQuery(query, null);
+        String queryStart = SELECT_FEED_ITEMS_AND_MEDIA_WITH_DESCRIPTION
+                + " WHERE " + queryFeedId + " AND (";
+        StringBuilder sb = new StringBuilder(queryStart);
+
+        for (int i = 0; i < queryWords.length; i++) {
+            sb
+                    .append(KEY_DESCRIPTION + " LIKE '%").append(queryWords[i])
+                    .append("%' OR ")
+                    .append(KEY_CONTENT_ENCODED).append(" LIKE '%").append(queryWords[i])
+                    .append("%' OR ")
+                    .append(KEY_TITLE).append(" LIKE '%").append(queryWords[i])
+                    .append("%' ");
+
+            if (i != queryWords.length - 1) {
+                sb.append("OR ");
```

Comment:
I think this should be "AND" (and use "()" around the "OR" parts). The search feature should only return episodes that contain all keywords, not only one of them